datafusion/core/src/dataframe.rs

```diff
@@ -437,19 +416,27 @@ impl DataFrame {
         ))];
         for field in original_schema_fields {
             let mut array_datas = vec![];
-            for record_batch in describe_record_batch.iter() {
-                // safe unwrap since aggregate record batches should have at least 1 record
-                let column = record_batch.get(0).unwrap().column_by_name(field.name());
-                match column {
-                    Some(c) => {
-                        if field.data_type().is_numeric() {
-                            array_datas.push(cast(c, &DataType::Float64)?);
-                        } else {
-                            array_datas.push(cast(c, &DataType::Utf8)?);
+            for result in describe_record_batch.iter() {
+                match result {
+                    Ok(df) => {
+                        let record_batch = df.clone().collect().await?;
+                        if record_batch.len() == 1 {
```

Comment:
Thiking, if the code can be simplified a bit, now the code looks overloaded with conditions.

Reply:
Thanks, I made some changes. If you have time, could you please help with another code review?

Reply:
> Thiking, if the code can be simplified a bit, now the code looks overloaded with conditions.

Thanks, I made some changes. If you have time, could you please help with another code review?